src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_TupleCreationExpression.cs

```diff
@@ -13,8 +9,80 @@ internal partial class LocalRewriter
+        {
+            int remainder;
+            int chainLength = TupleTypeSymbol.NumberOfValueTuples(rewrittenArguments.Length, out remainder);
+
+            NamedTypeSymbol underlyingTupleType = ((TupleTypeSymbol)node.Type).UnderlyingTupleType;
+            ArrayBuilder<NamedTypeSymbol> underlyingTupleTypeChain = GetUnderlyingTypeChain(underlyingTupleType);
+
+            var currentCreation = default(BoundObjectCreationExpression);
+
+            {
+                // make a creation expression for the smallest type
+                ImmutableArray<BoundExpression> ctorArguments = ImmutableArray.Create(rewrittenArguments, (chainLength - 1) * TupleTypeSymbol.chunk7, remainder);
+
+                var ctor = (MethodSymbol)Binder.GetWellKnownTypeMember(_compilation, TupleTypeSymbol.GetTupleCtor(remainder), _diagnostics, syntax: node.Syntax);
+                if ((object)ctor == null)
+                {
+                    underlyingTupleTypeChain.Free();
+                    return node;
+                }
+
+                MethodSymbol constructor = ctor.AsMember(underlyingTupleTypeChain.Pop());
+                currentCreation = new BoundObjectCreationExpression(node.Syntax, constructor, ctorArguments);
+            }
+
+            // make successively larger creation expressions containing the previous one
+            for (int linkIndex = chainLength - 2; linkIndex >= 0; linkIndex--)
+            {
+                ImmutableArray<BoundExpression> ctorArguments = ImmutableArray.Create(rewrittenArguments, linkIndex * TupleTypeSymbol.chunk7, TupleTypeSymbol.chunk7).Add(currentCreation);
+
+                var ctor = (MethodSymbol)Binder.GetWellKnownTypeMember(_compilation, TupleTypeSymbol.GetTupleCtor(TupleTypeSymbol.chunk8), _diagnostics, syntax: node.Syntax);
+                if ((object)ctor == null)
+                {
+                    underlyingTupleTypeChain.Free();
+                    return node;
+                }
+
+                MethodSymbol constructor = ctor.AsMember(underlyingTupleTypeChain.Pop());
+                currentCreation = new BoundObjectCreationExpression(node.Syntax, constructor, ctorArguments);
+            }
+
+            underlyingTupleTypeChain.Free();
+            return currentCreation;
+        }
+
+        private ArrayBuilder<NamedTypeSymbol> GetUnderlyingTypeChain(NamedTypeSymbol underlyingTupleType)
+        {
+            var underlyingTupleTypeChain = ArrayBuilder<NamedTypeSymbol>.GetInstance();
+            NamedTypeSymbol currentType = underlyingTupleType;
+
+            while (true)
+            {
+                underlyingTupleTypeChain.Add(currentType);
+                if (currentType.TypeArguments.Length == TupleTypeSymbol.chunk8)
+                {
+                    currentType = (NamedTypeSymbol)currentType.TypeArguments[TupleTypeSymbol.chunk7];
```

Comment:
TypeArguments is banned inside compiler, it looks like we missed this in analyzer. TypeArgumentsNoUseSiteDiagnostics should be used instead if you are not interested in use-site diagnostics, and TypeArgumentsWithDefinitionUseSiteDiagnostics if you are. The goal to make this choice explicit.